src/main/java/net/fabricmc/loom/configuration/providers/mappings/NoOpIntermediateMappingsProvider.java

```diff
@@ -37,11 +40,19 @@
  * A bit of a hack, creates an empty intermediary mapping file to be used for mc versions without any intermediate mappings.
  */
 public abstract class NoOpIntermediateMappingsProvider extends IntermediateMappingsProvider {
-	private static final String HEADER = "tiny\t2\t0\tofficial\tintermediary";
+	private static final String HEADER_OFFICIAL_MERGED = "tiny\t2\t0\tofficial\tintermediary";
+	private static final String HEADER_OFFICIAL_SPLIT = "tiny\t2\t0\tintermediary\tclientOfficial\tserverOfficial\t";
+
+	@Inject
+	public NoOpIntermediateMappingsProvider() {
+		getIsOfficialSplit().finalizeValueOnRead();
+	}
+
+	public abstract Property<Boolean> getIsOfficialSplit();
```

Comment:
It might be better to include the minecraft release time in `IntermediateMappingsProvider` so other providers can decide to use the new namespaces?

Reply:
At the moment it's pretty much tied to being before or after 1.3, if we want it to be more flexible this'll have to be changed in several other places too. All the mapped minecraft providers, the intermediate mappings service, the mappings merger. I'm not sure how straightforward it'd be but I'll have a look.